File: frontend/src/types/levels.ts
export type LevelKind =
  | "support"
  | "resistance"
  | "target_up"
  | "target_down"
  | "stop_zone"
  | "info";

export type TechnicalLevel = {
  kind: LevelKind;
  label: string;
  price: number | null;
  distance_percent: number | null;
  reason: string;
};

export type TechnicalLevelsResponse = {
  instrument_id: number;
  timeframe: string;
  last_close: number | null;
  atr: number | null;
  atr_percent: number | null;
  lookback: number;
  levels: TechnicalLevel[];
  summary: string;
  generated_at: string;
  message?: string | null;
};

File: backend/app/schemas/levels.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel

LevelKind = Literal["support", "resistance", "target_up", "target_down", "stop_zone", "info"]


class TechnicalLevel(BaseModel):
    kind: LevelKind
    label: str
    price: float | None
    distance_percent: float | None
    reason: str


class TechnicalLevelsResponse(BaseModel):
    instrument_id: int
    timeframe: str
    last_close: float | None
    atr: float | None
    atr_percent: float | None
    lookback: int
    levels: list[TechnicalLevel]
    summary: str
    generated_at: datetime
    message: str | None = None
